subprojects/core/src/integTest/groovy/org/gradle/initialization/CommandLineArgDeprecationIntegrationTest.groovy

```diff
@@ -25,11 +25,12 @@ class CommandLineArgDeprecationIntegrationTest extends AbstractIntegrationSpec {
     private static final String RECOMPILE_SCRIPTS_MESSAGE = '--recompile-scripts has been deprecated and is scheduled to be removed in Gradle'
 
     private static final String NO_REBUILD_MESSAGE = '--no-rebuild/-a has been deprecated and is scheduled to be removed in Gradle'
+    private static final String NO_SEARCH_UPWARD_MESSAGE = '--no-search-upward/-u has been deprecated and is scheduled to be removed in Gradle'
 
     @Unroll
     def "deprecation warning appears when using #deprecatedArgs"() {
         when:
-        executer.expectDeprecationWarning()
+        executer.expectDeprecationWarning().requireGradleDistribution()
```

Comment:
⭕️  Why do we need a distribution for this?

Reply:
It looks like the deprecation message is emitted before we even capture standard output. That might be reason why we [batch up the deprecation messages in `StartParameterInternal`](https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/internal/StartParameterInternal.java#L44) in then [check them later](https://github.com/gradle/gradle/blob/master/subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/StartParamsValidatingActionExecuter.java#L65-L67)?

Reply:
Logging is set up [before](https://github.com/gradle/gradle/blob/ac48633eab60f328da4fd6a080c31dda43b60b91/subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/LauncherServices.java#L74) that, so it should work. Bo also removed the `requireGradleDistribution` from all the other warning tests, so I wonder why it wouldn't work in this one.

Reply:
Not quite sure. You can even see the deprecation message in the console output if you remove `requireGradleDistribution()`. Any other ideas why that could be?